client/network/sync/src/engine.rs

```diff
@@ -791,10 +1065,14 @@ where
 			}
 		}
 
-		// poll `ChainSync` last because of a block announcement was received through the
-		// event stream between `SyncingEngine` and `Protocol` and the validation finished
-		// right after it as queued, the resulting block request (if any) can be sent right away.
-		while let Poll::Ready(result) = self.chain_sync.poll(cx) {
+		// Drive `ChainSync`.
+		while let Poll::Ready(()) = self.chain_sync.poll(cx) {}
+
+		// Poll block announce validations last, because if a block announcement was received
+		// through the event stream between `SyncingEngine` and `Protocol` and the validation
+		// finished right after it is queued, the resulting block request (if any) can be sent
+		// right away.
+		while let Poll::Ready(result) = self.poll_block_announce_validation(cx) {
```

Comment:
Like if this was just `self.block_announce_validator.poll_next_unpin(cx)`, it'd be easy to transfer it to use `tokio::select!` later on when the `SyncingEngine` is fully refactored.

Reply:
Done